src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -223,6 +227,23 @@ const AppWrapper = (props: { children: ReactNode }) => {
       }
     }
 
+    const multiSigGroups: MultiSigGroup[] = (
+      await Promise.all(
+        storedIdentifiers.map(async (identifier) => {
+          if (identifier.groupMetadata) {
+            const groupId = identifier.groupMetadata.groupId;
+            const connections =
+              await Agent.agent.connections.getMultisigLinkedContacts(groupId);
+            return {
+              groupId,
+              connections,
+            };
+          }
+          return null;
+        })
+      )
+    ).filter((group): group is MultiSigGroup => group !== null);
```

Comment:
This is too expensive and unnecessary - we just need it while we are creating the identifier. Otherwise we will be loading all these linked contacts for every multi-sig we've ever created.

Can we:
- Keep the cache empty in Redux - and only have support for "one" multisig list of connections at a time,
- When creating a multi-sig or opening a previous one that isn't fully initiated, whenever we need to call `getMultisigLinkedContacts` for the first time, populate it,
- Clear Redux when the component unmounts from the DOM.

Reply:
That being said - if we end up with a more async approach, it might be good to be in Redux so the AppWrapper can easily update it